onnxruntime/core/so/cxa_demangle.cc

```diff
@@ -0,0 +1,39 @@
+#if USE_DUMMY_EXA_DEMANGLE
+
+extern "C" {
+#include <stddef.h>
+#include <limits.h>
+
+/**
+ * An dummy implementation of __cxa_demangle https://itanium-cxx-abi.github.io/cxx-abi/abi.html#demangler
+ * It is referred in https://github.com/llvm/llvm-project/blob/d09d297c5d/libcxxabi/src/cxa_default_handlers.cpp#L53
+ * However it contribute a large percentage of binary size in minial build. To avoid it, the LLVM user can compile
+ * libc++abi.a with LIBCXXABI_NON_DEMANGLING_TERMINATE defined, see https://reviews.llvm.org/D88189
+ * But this make the compilation process extremely complex. Here we provide a dummy __cxa_demangle. It always copy
+ * the mangled name to output buffer. The required allocation behavior in Itanium C++ ABI will result a failure status,
+ * a.k.a, not implemented.
+ */
+char* __cxa_demangle(const char* mangled_name, char* output_buffer, size_t* length, int* status) {
+  if (!status) return NULL;
+  if (!output_buffer) {
+    *status = -1;
```

Comment:
nit: given we didn't do the memory allocation should this return -3 for invalid argument instead?

would be nice to have some names to the error codes for readability - either via an enum or some constants.

Reply:
In the API doc, the buffer can be null. From the caller's PoV, it is valid, so I think it is better to use `memory allocation failure` in this case.

Reply:
If it's null isn't the implementation meant to allocate, which would potentially make returning an error about memory allocation valid? 

>If buf is a null pointer, __cxa_demangle allocates a new buffer with malloc. It stores the size of the buffer in *n, if n is not NULL.